packages/formatters/src/github-actions.ts

```diff
@@ -0,0 +1,42 @@
+import { relative } from '@stoplight/path';
+import { DiagnosticSeverity, Dictionary } from '@stoplight/types';
+import { Formatter } from './types';
+
+const OUTPUT_TYPES: Dictionary<string, DiagnosticSeverity> = {
+  [DiagnosticSeverity.Error]: 'error',
+  [DiagnosticSeverity.Warning]: 'warning',
+  [DiagnosticSeverity.Information]: 'notice',
+  [DiagnosticSeverity.Hint]: 'notice',
+};
+
+type OutputParams = {
+  title?: string;
+  file: string;
+  col?: number;
+  endColumn?: number;
+  line?: number;
+  endLine?: number;
+};
+
+export const githubActions: Formatter = results => {
+  return results
+    .map(result => {
+      // GitHub Actions requires relative path for annotations, determining from working directory here
+      const file = relative(process.cwd(), result.source ?? '');
+      const params: OutputParams = {
+        title: result.code.toString(),
+        file,
+        col: result.range.start.character,
```

Comment:
doesn't Github's `endColumn` start at 1?
`IDiagnostic` that Spectral implements follows LSP where [positions](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#position) are 0-based, both the character and line.

```suggestion
        col: result.range.start.character + 1,
```

Reply:
Nice catch, both `line` and `col` start at 1 in GitHub Actions [^1].
Fixed in [77cddc0](https://github.com/stoplightio/spectral/pull/2508/commits/77cddc005d1b47f58e88cd9371c01f78665a1283).

[^1]: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-notice-message